cmd/loki-canary/main.go

```diff
@@ -112,15 +116,45 @@ func main() {
 	sentChan := make(chan time.Time)
 	receivedChan := make(chan time.Time)
 
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	logger = log.With(logger, "caller", log.Caller(3))
+
 	c := &canary{}
 	startCanary := func() {
 		c.stop()
 
 		c.lock.Lock()
 		defer c.lock.Unlock()
 
-		var err error
-		c.writer = writer.NewWriter(os.Stdout, sentChan, *interval, *outOfOrderMin, *outOfOrderMax, *outOfOrderPercentage, *size)
+		var (
+			err error
+			w   io.Writer
+		)
+
+		w = os.Stdout
+
+		if *push {
+			push, err := writer.NewPush(
+				*addr,
+				*tenantID,
+				*writeTimeout,
+				config.DefaultHTTPClientConfig,
+				*lName, *lVal,
+				*sName, *sValue,
+				tlsConfig,
+				*caFile,
+				*user, *pass,
+				log.NewLogfmtLogger(os.Stdout),
+			)
+			if err != nil {
+				_, _ = fmt.Fprintf(os.Stderr, "Unable to create writer for Loki, check config: %s", err)
```

Comment:
Why not use the logger here?

Reply:
Currently, before this PR, canary was never using proper logger and were writing directly to os.Stderr and os.Stdout (both in `main.go` and other components).

I just wanted to keep it consistent for this PR (focussed on only changes related to `push`) and do cleanup in separate PR as mentioned in the reviewer note
```
There will be follow up PR(s) to cleanup, particularly reader and comparitor component in terms of logging with proper logger. Rationale is to keep the changes small per PR to make it easy to review.
```